core/src/main/java/com/sequenceiq/cloudbreak/service/stack/StackImageService.java

```diff
@@ -93,7 +93,18 @@ public Image getCurrentImage(Stack stack) throws CloudbreakImageNotFoundExceptio
     public void changeImageCatalog(Stack stack, String imageCatalog) {
         try {
             Image currentImage = componentConfigProviderService.getImage(stack.getId());
+            ImageCatalog sourceImageCatalog = imageCatalogService.getImageCatalogByName(stack.getWorkspace().getId(), currentImage.getImageCatalogName());
+            if (imageCatalogService.isCustomImageCatalog(sourceImageCatalog)) {
+                throw new CloudbreakServiceException(
+                        String.format("Changing custom image catalog '%s' is not supported.", sourceImageCatalog.getName()));
+            }
+
             ImageCatalog targetImageCatalog = imageCatalogService.getImageCatalogByName(stack.getWorkspace().getId(), imageCatalog);
+            if (imageCatalogService.isCustomImageCatalog(targetImageCatalog)) {
+                throw new CloudbreakServiceException(
+                        String.format("Custom image catalogs are not supported to replace the current '%s' image catalog.", sourceImageCatalog.getName()));
```

Comment:
This error message is misleading - a custom JSON image catalog is still a custom image catalog. There should be something like "only JSON catalogs are supported" or something like that.